src/Infrastructure/Identity/IdentityService.cs

```diff
@@ -74,7 +85,7 @@ public async Task<Result> DeleteUserAsync(string userId)
             return Result.Success();
         }
 
-        public async Task<Result> DeleteUserAsync(ApplicationUser user)
+        public async Task<Result> DeleteUserAsync(User user)
```

Comment:
Shouldn't there be another user synchronization after deletion (same as after adding new user)? Something like:
```csharp
    public async Task<Result> DeleteUserAsync(User user)
    {
        var result = await _userManager.DeleteAsync(user);
        if (result.Succeeded)
        {
            // Sync identity user and application user
            var userToDelete = _context.ApplicationUsers.FirstOrDefault(x => x.Id == user.Id);
            if (userToDelete != null)
            {
                _context.ApplicationUsers.Remove(userToDelete);
                await _context.SaveChangesAsync(CancellationToken.None);
            }     
        }
        return result.ToApplicationResult();
    }
```